salt/modules/pip.py

```diff
@@ -335,6 +335,21 @@ def _process_requirements(requirements, cmd, cwd, saltenv, user):
     return cleanup_requirements, None
 
 
+def _pass_env_vars_to_cmd(env_vars, cmd_kwargs):
```

Comment:
This function doesn't pass anything to a command, no command is being run here. We prefer that function names describe what they do.

I also don't like that we force one to pass through a dictionary to be modified, it makes the function less flexible. In my opinion this function should return a dictionary, and its result then added to cmd_kwargs separately. This also makes unit testing easier, FWIW.

I think this function can be entirely replaced with something like this:

```python
def _format_env(env_vars):
    ret = {}
    if env_vars:
        try:
            for key, val in six.iteritems(env_vars):
                if not isinstance(val, six.string_types):
                    val = str(val)
                ret[key] = val
        except AttributeError:
            raise CommandExecutionError(
                'env_vars {0} is not a dictionary'.format(env_vars))
    return ret
```
Then lines 821 and 1030 can be simply changed to:

```python
cmd_kwargs.update(_format_env(env_vars))
```

Reply:
Thanks for looking at this @terminalmage . I have made the changes you pointed out.